pandera/backends/pyspark/container.py

```diff
@@ -80,6 +80,30 @@ def _column_checks(
 
         return check_obj
 
+    @validate_scope(scope=ValidationScope.DATA)
+    def _data_checks(
```

Comment:
A separate function was defined to run `DATA`-related validations when this validation depth is enabled, given that the existing one (`_schema_checks`) is designated to `SCHEMA`-related validations.

Reply:
Great @filipeo2-mck! I like how we're catching the SchemaError raised by the `unique` method, but I think it's a bit weird that we're rasing a Schema error when it's potentially nothing wrong with the schema. @cosmicBboy, does it make sense to introduce a `DataError` in `pandera.errors` to make sure we distigish between different type of errors, or is there any reason that we don't want it?

Reply:
Semantically, what does the `SchemaError` in `pandera.errors` object stands for?
- the segregation between two available validation types (**schema** vs data validations)? If this is the meaning, we should probably add data-related `DataError` exceptions to this namespace, to be specific about the kind of issue being raised.
- the Pandera objects or components, at a macro level (a **schema**, some data - the df, the declared checks, the existing columns...)? If this is the meaning, I see no issues about calling it `SchemaError`.

Reply:
Semantically, `SchemaError(s)` stand for anything that's wrong with the data or metadata of a validated object. That includes `metadata` (column names, types, etc), and `data` (actual data values contained in the object).

I think for clarity we should rename `ValidationScope.SCHEMA` to `ValidationScore.METADATA` to clarify the difference in pandera (I understand that the term Schema often refers to what I'm calling `metadata` here i.e. columns and their types, but pandera takes a slightly broader view of what a schema is).

> but I think it's a bit weird that we're rasing a Schema error when it's potentially nothing wrong with the schema. @cosmicBboy, does it make sense to introduce a DataError in pandera.errors to make sure we distigish between different type of errors, or is there any reason that we don't want it?

I'm not sure what the concern is here: if the `unique` method is the function that raises a `SchemaError` that's caught elsewhere in the validation pipeline, doesn't that mean, by definition, that there's something wrong with the data?